ClientTests/HistoryDeletionUtilityTests.swift

```diff
@@ -0,0 +1,260 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+import Shared
+
+@testable import Client
+@testable import Storage
+
+// This file has two built in assumptions:
+// 1. HistoryHighlights writes&deletes data correctly
+// 2. History writes&deletes data correctly
+// These basic cases are not tested here as they are tested in
+// `HistoryHighlightsManagerTests` and `TestHistory` respectively
+class HistoryDeletionUtilityTests: XCTestCase {
+    private struct SiteElements {
+        let domain: String
+        let path: String
+
+        init(domain: String, path: String = "") {
+            self.domain = domain
+            self.path = path
+        }
+    }
+
+    private var profile: MockProfile!
+    private var deletionUtility: HistoryDeletionUtility!
+
+    override func setUp() {
+        super.setUp()
+
+        profile = MockProfile(databasePrefix: "historyDeletion_tests")
+        profile._reopen()
+        deletionUtility = HistoryDeletionUtility(with: profile)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+
+        deletionUtility = nil
+        profile._shutdown()
+        profile = nil
+    }
+
+    func testEmptyRead() {
+        emptyDB()
+        assertDBIsEmpty()
+    }
+
+    func testSingleDataExists() {
+        emptyDB()
+
+        let testSites = [SiteElements(domain: "mozilla")]
+        populateDBHistory(with: testSites)
+
+        assertDBStateFor(testSites)
+    }
+
+    func testDeletingSingleItem() {
+        emptyDB()
+
+        let expectation = expectation(description: "HistoryDeletionUtilityTest")
+
+        let testSites = [SiteElements(domain: "mozilla")]
+        populateDBHistory(with: testSites)
+
+        let siteEntry = createWebsiteFor(domain: "mozilla", with: "")
+        deletionUtility.delete([siteEntry.url]) { result in
+            XCTAssertTrue(result)
+            self.assertDBIsEmpty()
+            expectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 30, handler: nil)
```

Comment:
30 sec is a really long time for unit tests. Is it really that long?

Reply:
So I tested it and it's not. But sometimes it is. I originally had it at 5 and it would fail half the time. But with it at 30 it seems to always be good. I'd love to make this better if you have a suggestion. :)